src/js/angular/repositories/controllers.js

```diff
@@ -715,6 +692,14 @@ function EditRepositoryCtrl($scope, $routeParams, toastr, $repositories, $locati
                             $scope.ontopRepoFileNames[key] = $scope.repositoryInfo.params[key].value;
                         }
                     });
+
+                    if ($scope.repositoryInfo.type === ONTOP_TYPE) {
+                        $scope.selectedDriver = Object.assign({},DEFAULT_SELECTED_DRIVER);
+                        Promise.resolve(getSupportedDriversData($scope, RepositoriesRestService, toastr))
```

Comment:
As long as `getSupportedDriversData` returns a promise, why do you wrap it in a promise again?